presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java

```diff
@@ -428,6 +428,7 @@ private QueryError toQueryError(ExecutionFailureInfo executionFailureInfo)
                     null,
                     errorCode.getCode(),
                     errorCode.getName(),
+                    executionFailureInfo.getSemanticErrorCode().name(),
```

Comment:
This bug is the cause of the Travis CI failures. The `.name()` dereference fails with `NullPointerException` when there is no semantic error code.

Let's change `ExecutionFailureInfo` to use `Optional<SemanticErrorCode>` rather than a nullable value. Then this code can become
```java
executionFailureInfo.getSemanticErrorCode()
        .map(SemanticErrorCode::name)
        .orElse(null)
```
In the future, we should change the rest of the nullable fields in that class to use `Optional`.